app/src/main/java/de/tum/in/tumcampusapp/tumonline/TUMBarrierFreeRequest.java

```diff
@@ -0,0 +1,292 @@
+                        }
+                    }
+                } catch (JSONException e) {
+                    Utils.log(e);
+                }
+
+                return buildingsToGps;
+            }
+
+            @Override
+            protected void onPostExecute(List<Map<String, String>> result) {
+                if (!isOnline) {
+                    listener.onNoInternetError();
+                    return;
+                }
+                if (result == null) {
+                    listener.onFetchError(context.getString(R.string.empty_result));
+                    return;
+                }
+
+                LocationManager locationManager = new LocationManager(context);
+                String buildingID = locationManager.getBuildingIDFromCurrentLocation(result);
+
+                if (buildingID == null || buildingID.equals("") || buildingID.equals("null")){
+                    listener.onFetchError(context.getString(R.string.empty_result));
+                    return;
+                }
+
+                String url = API_URL_Nerby_FACILITIES + "/" + buildingID;
+                fetchListOfFacilities(context, listener, url);
+            }
+
+        };
+
+        facilitiesBackgroundTask.execute();
+    }
+
+    public List<BarrierfreeContact> fetchResponsiblePersonList() {
+
+        String url = API_URL_RESPONSIBLEPERSON;
+        Optional<JSONArray> jsonArray = net.downloadJsonArray(url, CacheManager.VALIDITY_ONE_MONTH, false);
+
+        List<BarrierfreeContact> contactsList = new ArrayList<>();
+        try {
+            if (jsonArray.isPresent()) {
+                JSONArray arr = jsonArray.get();
+                for (int i = 0; i < arr.length(); i++) {
+                    JSONObject obj = arr.getJSONObject(i);
+
+                    String name = obj.getString(KEY_PERSON_NAME);
+                    String phone = obj.getString(KEY_PERSON_TELEPHONE);
+                    String email = obj.getString(KEY_PERSON_EMAIL);
+                    String faculty = obj.getString(KEY_PERSON_FACULTY);
+                    String tumID = obj.getString(KEY_PERSON_TUM_ID);
+
+                    if (!name.equals("null")){
+                        BarrierfreeContact contact = new BarrierfreeContact();
+                        contact.setName(name);
+                        contact.setFaculty(faculty);
+                        contact.setPhone(phone);
+                        contact.setEmail(email);
+                        contact.setTumonlineID(tumID);
+
+                        contactsList.add(contact);
+                    }
+
+                }
+            }
+        } catch (JSONException e) {
+            Utils.log(e);
+        }
+
+        return contactsList;
+    }
+
+    public List<BarrierfreeMoreInfo> fetchMoreInfoList(){
+        String url = API_URL_MORE_INFO;
+        Optional<JSONArray> jsonArray = net.downloadJsonArray(url, CacheManager.VALIDITY_ONE_MONTH, false);
```

Comment:
I just know what Retrofit is by Googling....
Sorry about that. I will change them.